### DISCUSSION TITLE
Using temp vectors

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#45] [General]
Some methods take Jolt vectors for initialization of some objects. For example, when creating a body creation settings object, we need to provide position/rotation vectors:
```js
Jolt.BodyCreationSettings(shape, position, rotation, motionType, layer);
```

Here, we obviously want to keep the shape. Motion type and layer are scalars, so can be ignored. How about position/rotation? When can those be destroyed/reused?

Assuming those are `new Jolt.Vec3()`, can they be reused immediately to generate a settings object for another body? Want to check if we need to allocate new vectors, or they are copied internally and we can reuse immediately.

**Comment by jrouwe** [comment]
They are copied internally, so you can reuse them.
